assets/js/components/setup/setup-wrapper.js

```diff
@@ -24,53 +24,25 @@ import { delay } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { withFilters } from '@wordpress/components';
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment, useCallback } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import Header from '../header';
 import Link from '../link';
 import HelpLink from '../help-link';
 import { getSiteKitAdminURL } from '../../util';
+import { STORE_NAME as CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+const { useSelect } = Data;
 
-class BaseComponent extends Component {
-	render() {
-		const { children } = this.props;
-		return (
-			<Fragment>
-				{ children }
-			</Fragment>
-		);
-	}
-}
-
-class SetupWrapper extends Component {
-	constructor( props ) {
-		super( props );
-
-		const { moduleToSetup } = global._googlesitekitLegacyData.setup;
-		this.state = {
-			currentModule: moduleToSetup,
-		};
-	}
-
-	static loadSetupModule( slug ) {
-		// Disabled because this rule doesn't acknowledge our use of the variable
-		// as a component in JSX.
-		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
-		const FilteredModuleSetup = withFilters( `googlesitekit.ModuleSetup-${ slug }` )( BaseComponent );
-
-		return (
-			<FilteredModuleSetup
-				finishSetup={ SetupWrapper.finishSetup }
-				onSettingsPage={ false }
-				isEditing={ true }
-			/>
-		);
-	}
+export default function SetupWrapper() {
+	const settingsPageURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' ) );
+	const { moduleToSetup } = global._googlesitekitLegacyData.setup;
```

Comment:
This was explicitly not required as part of the ACs, but maybe it would be worth moving this legacy dependency out of this component and require it as a prop instead (`moduleSlug`). This would for example simplify the Storybook configuration so that we no longer need to mess with the global there which is error-prone.

We could instead do the `global._googlesitekitLegacyData.setup.moduleToSetup` lookup in the components using `SetupWrapper` and pass the value as prop, so that at least this component is fully refactored (let's not worry about the other ones):
* `DashboardSplashApp`
* `GoogleSitekitDashboard`
* `GoogleSitekitModule`